addons/mail/controllers/websocket.py

```diff
@@ -0,0 +1,17 @@
+from odoo.addons.bus.controllers.websocket import WebsocketController
+from odoo.http import request, route, SessionExpiredException
+
+
+class WebsocketControllerPresence(WebsocketController):
+    @route('/websocket/update_bus_presence', type='jsonrpc', auth='public', cors='*')
+    def update_bus_presence(self, inactivity_period, im_status_ids_by_model):
+        if 'is_websocket_session' not in request.session:
+            raise SessionExpiredException()
+        request.env['ir.websocket']._update_bus_presence(int(inactivity_period), im_status_ids_by_model)
+        return {}
+
+    def get_subscribe_data(self, channels, last):
+        subscribe_data = super().get_subscribe_data(channels, last)
+        if bus_target := request.env["ir.websocket"]._get_missed_presences_bus_target():
```

Comment:
I feel like this method is not needed anymore. It was there to allow override from the mail module. Now that it's not an issue anymore it's just a check on whether we have a guest or a non public user.

Reply:
This whole controller is not used anymore, isn't it?

The route `/websocket/update_bus_presence` is not called anywhere

Reply:
Those are dedicated Odoo.SH routes that are called from outside odoo.

Reply:
Got it.

About `_get_missed_presences_bus_target`, it is till used in multiple places, is it still not worth it to keep it as a method to avoid repetion?

Reply:
It wouldn't be an issue if we provided a `_dispatch_bus_presence` method as suggested here: https://github.com/odoo/odoo/pull/177611#discussion_r1820532109. But nevermind.